fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -5067,19 +5067,8 @@ Standalone<VectorRef<KeyValueRef>> generateOldFormatMutations(
 	for (int part = 0; part * CLIENT_KNOBS->MUTATION_BLOCK_SIZE < param2Concat.size(); part++) {
 		KeyValueRef backupKV;
 		// Assign the second parameter as the part
-		backupKV.value = param2Concat.substr(part * CLIENT_KNOBS->MUTATION_BLOCK_SIZE,
-		                                     std::min(param2Concat.size() - part * CLIENT_KNOBS->MUTATION_BLOCK_SIZE,
-		                                              CLIENT_KNOBS->MUTATION_BLOCK_SIZE));
-		// Write the last part of the mutation to the serialization, if the buffer is not defined
-		if (!partBuffer) {
-			// part = 0
-			wrParam1 << bigEndian32(part);
-			partBuffer = (uint32_t*)((char*)wrParam1.getData() + wrParam1.getLength() - sizeof(uint32_t));
-		} else {
-			// part > 0
-			*partBuffer = bigEndian32(part);
-		}
-		backupKV.key = wrParam1.toValue();
+		backupKV.value = getBackupValue(param2Concat, part);
+		backupKV.key = getBackupKey(wrParam1, &partBuffer, part);
```

Comment:
Need a temporary variable to holds the value (including its arena) returned by `getBackupKey()`. Otherwise,  `backupKV.key` will reference invalid memory.